integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/mutation/DeleteExecution.scala

```diff
@@ -127,12 +126,16 @@ object DeleteExecution {
           var result = List[(SegmentStatus, (SegmentUpdateDetails, ExecutionErrors))]()
           while (records.hasNext) {
             val ((key), (rowCountDetailsVO, groupedRows)) = records.next
+            val segmentId = key.substring(0, key.indexOf(CarbonCommonConstants.FILE_SEPARATOR))
```

Comment:
Is it possible that return -1 when run key.indexOf?

Reply:
Not possible as in above method it adds segmentid.